Sources/SwiftRefactor/ExpandEditorPlaceholder.swift

```diff
@@ -0,0 +1,347 @@
+public struct ExpandEditorPlaceholders: RefactoringProvider {
+  public static func refactor(syntax token: TokenSyntax, in context: Void) -> ExprSyntax? {
+    guard let placeholder = token.parent?.as(EditorPlaceholderExprSyntax.self),
+      let arg = placeholder.parent?.as(TupleExprElementSyntax.self),
+      let argList = arg.parent?.as(TupleExprElementListSyntax.self),
+      let call = argList.parent?.as(FunctionCallExprSyntax.self)
+    else {
+      return ExpandEditorPlaceholder.refactor(syntax: token)
+    }
+
+    guard let expanded = call.expandTrailingClosurePlaceholders(ifIncluded: arg) else {
+      return ExpandEditorPlaceholder.refactor(syntax: token)
+    }
+
+    guard let trailing = CallToTrailingClosures.refactor(syntax: expanded.expr, in: .init(startAt: argList.count - expanded.numClosures)) else {
+      return ExpandEditorPlaceholder.refactor(syntax: token)
+    }
+
+    return ExprSyntax(trailing)
+  }
+}
+
+extension FunctionTypeSyntax {
+  /// Return a closure expression for this function type, eg.
+  /// ```
+  ///   (_ someInt: Int) -> String
+  /// ```
+  /// would become
+  /// ```
+  /// { someInt in
+  ///   <#T##String#>
+  /// }
+  /// ```
+  fileprivate var closureExpansion: ClosureExprSyntax {
+    let closureSignature: ClosureSignatureSyntax?
+    if !arguments.isEmpty {
+      var args = arguments.dropLast().map { ClosureParamSyntax(name: $0.expansionNameToken(), trailingComma: .commaToken()) }
+      args.append(ClosureParamSyntax(name: arguments.last!.expansionNameToken()))
+      closureSignature = ClosureSignatureSyntax(input: .simpleInput(ClosureParamListSyntax(args)))
+    } else {
+      closureSignature = nil
+    }
+
+    // Single statement for the body - the placeholder-ed type if non-void and
+    // 'code' otherwise.
+    let ret = output.returnType.description
+    let placeholder: String
+    if ret == "Void" || ret == "()" {
+      placeholder = ExpandEditorPlaceholder.wrapInTypePlaceholder("code", type: "Void")
+    } else {
+      placeholder = ExpandEditorPlaceholder.wrapInTypePlaceholder(ret, type: ret)
+    }
+
+    let statementPlaceholder = EditorPlaceholderExprSyntax(
+      identifier: .identifier(placeholder)
+    )
+    let closureStatement = CodeBlockItemSyntax(
+      item: .expr(ExprSyntax(statementPlaceholder))
+    )
+
+    return ClosureExprSyntax(
+      leftBrace: .leftBraceToken(),
+      signature: closureSignature,
+      statements: CodeBlockItemListSyntax([closureStatement]),
+      rightBrace: .rightBraceToken()
+    )
+  }
+}
+
+extension TupleTypeElementSyntax {
+  /// Return a token to use as the parameter name in the expanded closure.
+  /// We prefer the argument name if there is one and it isn't a wildcard,
+  /// falling back to the label with the same conditions, and finally just the
+  /// placeholder-ed type otherwise.
+  fileprivate func expansionNameToken() -> TokenSyntax {
```

Comment:
This could just be a computed property, right?

Reply:
If you'd prefer 🤷

Reply:
I would, not super strongly though.